src/protocol/http/src/test/java/org/apache/jmeter/protocol/http/util/TestGraphQLRequestParamUtils.java

```diff
@@ -97,9 +102,24 @@ void testInvalidGraphQLContentType(String contentType) {
         assertFalse(GraphQLRequestParamUtils.isGraphQLContentType(contentType));
     }
 
-    @Test
-    void testToPostBodyString() throws Exception {
-        assertEquals(EXPECTED_POST_BODY, GraphQLRequestParamUtils.toPostBodyString(params));
+    static Stream<Triple<String, JsonNode, JsonNode>> postBodyFieldNameAndJsonNodes() throws Exception {
+        final JsonNode expectedPostBodyJson = objectMapper.readTree(EXPECTED_POST_BODY);
+        final JsonNode actualPostBodyJson = objectMapper.readTree(
+                GraphQLRequestParamUtils.toPostBodyString(new GraphQLRequestParams(OPERATION_NAME, QUERY, VARIABLES)));
+        return Stream.of(
+                Triple.of(GraphQLRequestParamUtils.OPERATION_NAME_FIELD, expectedPostBodyJson, actualPostBodyJson),
+                Triple.of(GraphQLRequestParamUtils.VARIABLES_FIELD, expectedPostBodyJson, actualPostBodyJson),
+                Triple.of(GraphQLRequestParamUtils.QUERY_FIELD, expectedPostBodyJson, actualPostBodyJson));
+    }
+
+    @ParameterizedTest
+    @MethodSource("postBodyFieldNameAndJsonNodes")
+    void testToPostBodyString(Triple<String, JsonNode, JsonNode> comparisonTriple) throws Exception {
+        final String fieldName = comparisonTriple.getLeft();
+        final JsonNode expectedNode = comparisonTriple.getMiddle();
+        final JsonNode actualNode = comparisonTriple.getRight();
```

Comment:
Please use three different arguments instead of `Triple`

Reply:
Done: ff3eba435347a3148332aa22a0a6d6d0e08646ff. Thanks!